spring-core/src/main/java/org/springframework/util/CollectionUtils.java

```diff
@@ -105,6 +107,30 @@ public static <K, V> LinkedHashMap<K, V> newLinkedHashMap(int expectedSize) {
 		return new LinkedHashMap<>(computeMapInitialCapacity(expectedSize), DEFAULT_LOAD_FACTOR);
 	}
 
+	/**
+	 * Instantiate a new {@link HashSet} with an initial capacity
+	 * that can accommodate the specified number of elements without
+	 * any immediate resize/rehash operations to be expected.
+	 * @param expectedSize the expected number of elements (with a corresponding
+	 * capacity to be derived so that no resize/rehash operations are needed)
+	 * @see #newLinkedHashSet(int)
+	 */
+	public static <E> HashSet<E> newHashSet(int expectedSize) {
+		return new HashSet<>(computeMapInitialCapacity(expectedSize), DEFAULT_LOAD_FACTOR);
+	}
+
+	/**
+	 * Instantiate a new {@link LinkedHashSet} with an initial capacity
+	 * that can accommodate the specified number of elements without
+	 * any immediate resize/rehash operations to be expected.
+	 * @param expectedSize the expected number of elements (with a corresponding
+	 * capacity to be derived so that no resize/rehash operations are needed)
+	 * @see #newHashSet(int)
+	 */
+	public static <E> LinkedHashSet<E> newLinkedHashSet(int expectedSize) {
+		return new LinkedHashSet<>(computeMapInitialCapacity(expectedSize), DEFAULT_LOAD_FACTOR);
+	}
+
 	private static int computeMapInitialCapacity(int expectedSize) {
```

Comment:
```suggestion
	private static int computeInitialCapacity(int expectedSize) {
```

Probably better to rename this method since it's now used for maps and sets.

Reply:
I thought about that, but noticed that [Guava](https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/Sets.java#L262) and the [JDK](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/HashSet.java#L401) both refer to map capacity helpers in their versions of this helper. It sort of makes sense I guess since it ends up being backed by a Map. Don't really have a strong opinion so I'll make your suggested change.

Reply:
> It sort of makes sense I guess since it ends up being backed by a Map.

True. I just thought the new name would require less cognitive overhead for anyone reading the code. Though, to be honest, it's an internal helper method. So it doesn't really matter. 

In any case, thanks for making the change.

On a side note, I just learned that there's a new `HashSet.newHashSet()` factory method in Java 19. So thanks for that. 👍 

Though we still have a Java 17 baseline.